packages/fetchai/skills/simple_aggregation/behaviours.py

```diff
@@ -47,19 +48,26 @@ def __init__(self, **kwargs: Any) -> None:
         search_interval = cast(
             float, kwargs.pop("search_interval", DEFAULT_SEARCH_INTERVAL)
         )
+        self._max_soef_registration_retries = kwargs.pop(
+            "max_retries", DEFAULT_MAX_SOEF_REGISTRATION_RETRIES
+        )  # type: int
         super().__init__(tick_interval=search_interval, **kwargs)
 
+        self.failed_registration_msg = None  # type: Optional[OefSearchMessage]
+        self._nb_retries = 0
+
     def setup(self) -> None:
         """Implement the setup for the behaviour."""
         self._register_agent()
-        self._register_service_personality_classification()
 
     def act(self) -> None:
         """
         Implement the act.
 
         :return: None
         """
+        self._retry_failed_registration()
+
         strategy = cast(AggregationStrategy, self.context.strategy)
```

Comment:
this doesn't look right. why the reregistration all the time?

Reply:
are you referring to the `self._retry_failed_registration()` at the beginning of the new `act()`?
If so, this just reregisters if there is a registration failure: `if self.failed_registration_msg is not None:` at the top of `_retry_failed_registration`.

Reply:
I am referring to 71-81.

Reply:
71-81 sends search requests (it is from existing code) not registration request.